### ISSUE TITLE
The debug of website e.g. : "ProjectX_sample" in "root_VS2013" fails.

### ISSUE THREAD

**Issue by yusukemaegawa** [OpenTouryoProject/OpenTouryoTemplates#12]
Until Visual Studio 2012, I could set the value of virtual path.
Therefore, the url of page includes the value of virtual path.

>e.g.
>http://localhost:9999/ProjectX_sample/Aspx/start/menu.aspx

But, in Visual Studio 2013, I cannot set the value of virtual path.
Therefore, the url of page does not include the value of virtual path.

>e.g.
>http://localhost:9999/Aspx/start/menu.aspx

In some code, the urls including the value of virtual path are still remaining.
Because of this reason, the debug of "ProjectX_sample" fails.

**Comment by daisukenishino2** [comment]
@yusukemaegawa 

The following three solutions can be considered.
- Using the IIS to set a virtual path. (The current cope.)
- Using the IISExpress to set a virtual path.
- Use a relative path.
- Be changed to use the WebApplication.

### Using the IIS to set a virtual path.
This is the [current cope](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/master/root_VS2013/Readme.md).

### Using the IISExpress to set a virtual path.
There is a maybe needed to change the folder structure.
This change will cause problem that it will be difficult to merge operations.

### Use a relative path.
I like virtual path, because It's better in the following points.

If you move a files in case of  using virtual path,
then it is easy to identify modifications range.

### Be changed to use the WebApplication.
Virtual path can be set also at VS2013 by using the WebApplication.

But, I like that "copy the assemblies to the Bin folder" than "reference settings".
This method can be used only on the WebSite.
Current build process is in accordance with this method.

### Correspondence
I think that the changes to the WebApplication is the most realistic at the moment.

**Comment by yusukemaegawa** [comment]
I think that [ResolveUrl](https://msdn.microsoft.com/en-us/library/system.web.ui.control.resolveurl(v=vs.110).aspx) and [ResolveClientUrl](https://msdn.microsoft.com/en-us/library/system.web.ui.control.resolveclienturl(v=vs.110).aspx) are usable in using relative path.

# Example using ResolveUrl
## Present state
```c#
● サンプル（全層結合テスト）<br />
　・ <a href="/ProjectX_sample/Aspx/sample/crud/sampleScreen.aspx">ノーマル</a><br />
　・ <a href="/ProjectX_sample/Aspx/sample/crud/sampleScreen_cc.aspx">通信制御</a><br />
<br />
```

## Modified state
```c#
● サンプル（全層結合テスト）<br />
　・ <a href="<%= this.ResolveUrl("~/Aspx/sample/crud/sampleScreen.aspx") %>">ノーマル</a><br />
　・ <a href="<%= this.ResolveUrl("~/Aspx/sample/crud/sampleScreen_cc.aspx") %>">通信制御</a><br />
<br />
```

# Example using ResolveClientUrl
## Present state
```c#
<script type="text/javascript" src="/ProjectX_sample/Framework/js/common.js"></script>
<script type="text/javascript" src="/ProjectX_sample/Framework/js/ie_key_event.js"></script>

<link rel="stylesheet" href="/ProjectX_sample/Css/style.css" type="text/css"/>
```

## Modified state
```c#
<script type="text/javascript" src="<%= this.ResolveClientUrl("~/Framework/js/common.js") %>"></script>
<script type="text/javascript" src="<%= this.ResolveClientUrl("~/Framework/js/ie_key_event.js") %>"></script>

<link rel="stylesheet" href="<%= this.ResolveClientUrl("~/Css/style.css") %>" type="text/css"/>
```

I think that it becomes unnecessary to describe a pass including the virtual directory name ("ProjectX_sample") for a program by using ResolveUrl, ResolveClientUrl.
As described above, please renew a pass beginning in "/ProjectX_sample" using ResolveUrl, ResolveClientUrl.
(It should be necessary to modify BaseController and Custom Controls.)

# Goal
* You can run the ASP.NET Sample of the "same" code by use of Visual Studio 2012 or Visual Studio 2013.
    * The error by the difference of the pass do not occur.
    * The dialog is displayed normally.
    * Screen transition work normally.
    * The images or the icons are displayed normally.

**Comment by daisukenishino** [comment]
# Added Target
Also the following Web site has the same problem.
In addition, The client side program can not be changed dynamically according to the URL of the server.

## ServiceInterface
- ASPNETWebService
- WCFService

## WSClient_sample/
- WSClientSL_sample
- WSClientWinStore_sample

**Comment by daisukenishino** [comment]
@SandeepNayak077 

Please describe to this input column the contents of today's meeting as a specification.

**Comment by SandeepNayak077** [comment]
<h1>Specification</h1>

Dynamically set the URL of the server in the following files:
- [app.config](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/develop/root_VS2013/programs/C%23/Samples/WebApp_sample/ProjectX_sample/app.config#L23) 
- [common.js](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/develop/root_VS2013/programs/C%23/Samples/WebApp_sample/ProjectX_sample/Framework/Js/common.js#L458)
- *.aspx.cs ([Case-1](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/develop/root_VS2013/programs/C%23/Samples/WebApp_sample/ProjectX_sample/Aspx/testFxLayerP/nest/testScreen1nest.aspx.cs#L667) and [Case-2](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/develop/root_VS2013/programs/C%23/Samples/WebApp_sample/ProjectX_sample/Aspx/testFxLayerP/nest/testScreen1nest.aspx.cs#L943))

The [ResolveUrl](https://msdn.microsoft.com/en-us/library/system.web.ui.control.resolveurl(v=vs.110).aspx) or the [ResolveClientUrl](https://msdn.microsoft.com/en-us/library/system.web.ui.control.resolveclienturl(v=vs.110).aspx) methods are not useful in the [app.config](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/develop/root_VS2013/programs/C%23/Samples/WebApp_sample/ProjectX_sample/app.config#L23) and [common.js](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/develop/root_VS2013/programs/C%23/Samples/WebApp_sample/ProjectX_sample/Framework/Js/common.js#L458) files.

To dynamically set the URL of the server in the above mentioned files, 
Investigate and modify the implementation of the framework components (BaseController and Custom Controls), if necessary.

We will provide required work around for this issue.

<h1>Solution</h1>
\-

**Comment by daisukenishino** [comment]
@SandeepNayak077 

#### Following url type should be supported.
Framework's API must support the following arguments.

- ASP.NET Web Project Paths  
https://msdn.microsoft.com/en-us/library/ms178116.aspx
  - Absolute path ("http://XXXX")
  - Virtual path ("/application_root_folder_name/XXXX")
  - Relative path ("/XXXX" or "../XXXX")
  - root-relative path ("~/XXXX")  
~ operator used to specify a root-relative path

If you will use ResolveUrl method in the Framework,
then you should note about the method's specification.

#### How to discover the modification target.
Please grep using the following keywords.
- "HTTP://"
- "/ProjectX_sample/"

For example, it will also be target of modify functions that  are using the following definitions.
https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/master/root_VS2013/files/resource/Xml/SCDefinition.xml

#### Policy of the modification.
And, please clarify the policy of the modification.
it should be described in the above solutions column.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

Sure we will do testing with above mentioned [URL types](https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/12#issuecomment-169950751).

/>And, please clarify the policy of the modification.
it should be described in the above solutions column.

Once the investigation and testing is completed,
We will write the description about the solution in the solution column.
After that we will take confirmation from @daisukenishino about solution.
Only after confirmation, we will do code commit and create pull request.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

We have investigated and found a solution to resolve url issue in the framework.
We have filled the description in the [above solution column](https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/12#issuecomment-169932650).

Please review and let us know.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

I checked [above solution column](https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/12#issuecomment-169932650).

The followings are "pointed out".

#### Please tell me how to resolve URL inside the common.js.
It is not described at the moment.

#### The following coding is easier.
```CS
if (url != null || url != "")
```
↓ ↓ ↓
```CS
if (!string.IsNullOrEmpty(url))
```
#### Do not be coding as follows ?
```CS
url = (HttpContext.Current.Handler as Page).ResolveUrl(url);
```
↓ ↓ ↓
```CS
url = this.ResolveUrl(url); // this is BaseController that inherit System.Web.UI.Page.
```
#### ResolveUrl method can handling all of the URL of the following ?
- Absolute path ("http://XXXX")
- Virtual path ("/application_root_folder_name/XXXX")
- Relative path ("/XXXX" or "../XXXX")
- root-relative path

**Comment by daisukenishino** [comment]
# Test method.
- /ProjectX_sample/ application should be able to be executed by F5 without performing the added procedures in [Readme.md of VS2013](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/master/root_VS2013/Readme.md).
- Also please check the functions that are relating modified parts in this time are working correctly by using /ProjectX_sample/ application.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

>Please tell me how to resolve URL inside the common.js.

We will soon add about this solution.
Currently, this issue is in testing phase.

>The following coding is easier.

Ok, we will use code as pointed above.

>Do not be coding as follows ?

No, it’s not possible.

<b>Why it’s not possible?</b>

The <b>ResolveServerUrl</b> method is implemented inside private <b>ScreenControl</b> class, which is nested inside abstract <b>BaseController</b> class.
So, we cannot implement the following code.
```
url = this.ResolveUrl(url);
```

<b>Why it is required to implement ResolveServerUrl method?</b>

We failed to resolve the URL issue in [this](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/master/root_VS2013/programs/C%23/Frameworks/Infrastructure/Framework/Presentation/BaseController.cs#L4161) screen transition control component.
To overcome this problem we have implemented <b>ResolveServerUrl</b> method.

>ResolveUrl method can handling all of the URL of the following ?

Yes, we have tested the ResolveUrl method with all the mentioned relative path.

Please confirm to continue the work.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

I have understood your corespondance method on all of the above, except for common.js.
Please continue to this work.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

We have updated the solution column in the [above comment](https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/12#issuecomment-169932650).

We have also added a "URL issue in the common.js file" column inside solution column.
Since the last R&D testing is failed, we are again investigating to resolve URL issue.
Once it is completed we will inform you for the confirmation.  

Please confirm.

**Comment by yusukemaegawa** [comment]
@daisukenishino 
@SandeepNayak077 

Do you consider the path of javascript or css in master page?

ex. [testBlankScreen.master](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/master/root_VS2013/programs/C%23/Samples/WebApp_sample/ProjectX_sample/Aspx/Common/testBlankScreen.master#L11)
```html
<script type="text/javascript" src="/ProjectX_sample/Framework/js/common.js"></script>
<script type="text/javascript" src="/ProjectX_sample/Framework/js/ie_key_event.js"></script>

<link rel="stylesheet" href="/ProjectX_sample/Css/style.css" type="text/css"/>
```

**Comment by SandeepNayak077** [comment]
@yusukemaegawa 

> Do you consider the path of javascript or css in master page?

Yes, we have already considered.
And this case has already been resolved with [this comment](https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/12#issuecomment-148618549). 

We have done the following modification of code in the xxxxx.master files:
```
<script type="text/javascript" src="<%= this.ResolveClientUrl("~/Framework/js/common.js") %>"></script>
<script type="text/javascript" src="<%= this.ResolveClientUrl("~/Framework/js/ie_key_event.js") %>"></script>

<link rel="stylesheet" type="text/css" href="<%= this.ResolveClientUrl("~/Css/style.css") %>">/>
```

<h4>New specification</h4>
We have written [new specification](https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/12#issuecomment-169932650) apart from the above issue. 
The solution column is only related to [this new specification](https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/12#issuecomment-169932650).

**Comment by SandeepNayak077** [comment]
@daisukenishino 

We have updated the [solution column](https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/12#issuecomment-169932650) with the description to resolve URL issue in common.js file.

Please confirm.

**Comment by yusukemaegawa** [comment]
@SandeepNayak077 

>We have written new specification apart from the above case. 
>The solution column is only related to this new specification.

Thank you for the rapid response.

>Please let us know if we are missing any part.

All right.
I will inform you if I find a problem.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

This is the information about the target.

##### Module
It has already been shown.
https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/12#issuecomment-154292428

##### Directory
is root_VS2013.

After that, there is a need to merge this modification to the following directory.
- root (Frameworks, ProjectX_sample, ASPNETWebService, WCFService, WSClientSL_sample)
- root_VS2012 (WSClientWinStore_sample)

**Comment by SandeepNayak077** [comment]
@daisukenishino 

We have updated the <b>URL issue in the common.js file</b> column in the above solution area.
Please review and let us know.

We are also requesting to close the [pull request-72](https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/72/).
We will send new pull request with the javascript changes.

**Comment by daisukenishino** [comment]
@kishoreandukuri 

>We will send new pull request with the javascript changes.

I see.

When you enclose text,
Please specify the type, like this.

\`\`\`XML
\`\`\`

**Comment by SandeepNayak077** [comment]
@daisukenishino 

>>This is the information about the target.

We have added <b>Traget framework</b> column in [this comment](https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/12#issuecomment-169932650).

>> When you enclose text,
>> Please specify the type, like this.

Sure, we will specify accordingly.
Thank you for the information.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

We did investigation to specify a relative URL in the Windows Store App sample project for [such code](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/master/root_VS2013/programs/C%23/Samples/WS_sample/WSClient_sample/WSClientWinStore_samples/WSClientWinStore_sample/Views/CRUDPage_jn.xaml.cs#L264). 
According to [this link](https://msdn.microsoft.com/en-us/library/ms131565(v=vs.110).aspx), we must not specify relative URI in the URI constructor for the Windows Store apps or the Portable Class Library projects.

So, we are finding any other alternatives like maintain URL paths in the configuration file, to resolve URL issue in WinStore_sample project. 
Once, we get an alternative we will report you.

please confirm.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

>to resolve URL issue in WinStore_sample project. 
Once, we get an alternative we will report you.

Thank you for your continuing support.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

> https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/12#issuecomment-180432211

We found the following a solution, to resolve URL issue in Windows store apps.

Add the following property in [this file](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/master/root_VS2013/programs/C%23/Samples/WS_sample/WSClient_sample/WSClientWinStore_samples/WSClientWinStore_sample/App.xaml#L10).

```XAML
<x:String x:Key="BaseUrl">http://localhost:9996/</x:String>
```  

However, the value of BaseUrl must be manually changed based on the application server.
For example:
- <b>VS Development Server:</b> http://localhost:9996/
- <b>IIS Server: </b>http://localhost:/WSClientSL_sample/

In the application page,
The BaseUrl must be concatenated with the relative path [here](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/master/root_VS2013/programs/C%23/Samples/WS_sample/WSClient_sample/WSClientWinStore_samples/WSClientWinStore_sample/Views/CRUDPage_jn.xaml.cs#L195).
```CS
string baseUrl = (string)App.Current.Resources["BaseUrl"];
WebRequest request = WebRequest.Create(new Uri(baseUrl + "WebService/ServiceForRt.svc/JSON", UriKind.Absolute));
```

Currently, I was finding a common location to access the BaseUrl.
```CS
string baseUrl = (string)App.Current.Resources["BaseUrl"];
```
So that it can be shared among other pages in the application like JSON and XML pages. 

Please let us about your opinion on the above solution.
If this is Ok, we will continue our investigation.

**Comment by daisukenishino** [comment]
@yusukemaegawa 

> If this is Ok, we will continue our investigation.

Please check the modification method.

@SandeepNayak077 

Please check, because target information is updated.
- https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/12#issuecomment-154292428
- https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/12#issuecomment-172820604

**Comment by yusukemaegawa** [comment]
@daisukenishino 
@SandeepNayak077 

I think that sample programs send request to web services three ways.

- using [CallController](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/master/root_VS2013/programs/C%23/Frameworks/Infrastructure/Framework/Transmission/CallController.cs) class
    - Developers should specify the url of web service in [TMProtocolDefinition.xml](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/master/root_VS2013/files/resource/Xml/TMProtocolDefinition.xml).
- using Service Reference
    - Developers should specify the url of web service in [the "Add Service Reference" dialog box](https://msdn.microsoft.com/library/bb628649.aspx).
- using WebRequest class
    - Developers should specify the url of web service in [the argument of WebRequest.Create method](https://github.com/OpenTouryoProject/OpenTouryoTemplates/blob/master/root_VS2013/programs/C%23/Samples/WS_sample/WSClient_sample/WSClientWinStore_samples/WSClientWinStore_sample/Views/CRUDPage_jn.xaml.cs#L195).

> [#12 (comment)](https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/12#issuecomment-182219391)

I think that this countermeasure is effective in the way of "using WebRequest class".
Are there the effective methods for the ways of "using CallController class" and "using Service Reference"?

**Comment by daisukenishino** [comment]
@yusukemaegawa 
@SandeepNayak077 

- Correspondence of CallController class is required.  
  - To do this, there is a need to modify the implementation of CallController.
  - To change the path in the "TMProtocolDefinition.xml" file.
  - This can address to the all of clients that use the communication control function.
- Correspondence of WebRequest class is required.
   - REST Web API of Silverlight and Windows Store app is used this class.
- Service Reference of correspondence is not required.
   - SOPA Web API of Silverlight and Windows Store app is used this class.

This modification must also be carried out to root and root_VS2012(for Windows Store Apps) folder for future's merge.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

> Correspondence of CallController class is required.

We will plan and add task about this correspondence.

> To change the path in the " TMProtocolDefinition.xml" file.

Currently, in this file all values for URIs are represented as Absolute URI path.
This absolute URI path changes depending on the versions of visual studio (VS2010-VS2012 and VS2013-VS2015).
However, representation of relative path might be same.

So my question in this correspondence is,
What kind (either absolute or relative path) of URI value must be represented in the "TMProtocolDefinition.xml" file?

**Comment by daisukenishino** [comment]
@SandeepNayak077 

>What kind (either absolute or relative path) of URI value must be represented in the "TMProtocolDefinition.xml" file?

Thank you very much of exact answer.

Certainly, modification of the program was unnecessary because CallController uses In Absolute URI path. I think that this absolute URI path format is required to consider because only absolute URI path format is used in the TMProtocolDefinition.xml file.

I propose idea such as following.
- Copy the TMProtocolDefinition.xml as TMProtocolDefinition2.xml.
- Exclude the project name from the absolute URI path of TMProtocolDefinition.xml.
- Users use this function by switching these two files.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

Thank you for the compliment.

We have updated [this comment]() based on the above discssions.
Please check <b>Solution to the correspondence of CallController class</b> column.

> Exclude the project name from the absolute URI path of TMProtocolDefinition.xml.

Did small correction, excluded the project name from TMProtocolDefinition2.xml file instead of TMProtocolDefinition.xml file.
Because, the TMProtocolDefinition.xml file is unique for all other versions of frameworks. 
And while review, we will get single git difference like the TMProtocolDefinition2.xml is added.

please confirm.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

> Please check Solution to the correspondence of CallController class column.
Did small correction,

I’m torn between these.

- Excluded the project name from TMProtocolDefinition.xml  
Because virtual URI path doesn't have project name in the default from VS2013.
- Excluded the project name from TMProtocolDefinition2.xml  
Because check of difference of the merge work will be easy  
(Because TMProtocolDefinition.xml isn't changed).

But, this is a difficult to say which of the two is better.
Therefor, I want to adopt your proposal.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

Thank you for the confirmation.

**Comment by daisukenishino** [comment]
@yusukemaegawa 
@SandeepNayak077 

Target of the VB version had been leaked.
I will add to the target of the following.

- ProjectX_sample
- ServiceInterface
  - ASPNETWebService
  - WCFService

**Comment by SandeepNayak077** [comment]
@daisukenishino

> https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/12#issuecomment-189128554

Yes, we have not considered any project in the VB version.
Because all targeted projects are C# versions.

As we are planning for the next sprint today.
We will add this task in the plan.
Please confirm.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

>https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/12#issuecomment-189144144
As we are planning for the next sprint today.
We will add this task in the plan.

I kindly ask for your correspondence.
Please divided this correspondence into a some pull request.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

> https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/12#issuecomment-189158235
> Please divided this correspondence into a some pull request.

For the future pull requests we have planned to create with respect to the proects,
Like for silver light sample one pull request and another for windows store sample and so on.

Is this [pull request-73](https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/73) required to be divided?
Please let us know.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

>Is this pull request-73 required to be divided?

If this is the modifications within the scope of ProjectX_sample, division is not required.

>Like for silverlight sample one pull request and another for windows store sample and so on.

It is consistent with the our expectations.

>https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/12#issuecomment-172820604

Please divided as follows.

- ProjectX_sample
- ServiceInterface
- Silverlight sample
  - WSClientSL_sample
- Windows store sample (one pull request)
  - WSClientWinStore_sample
  - WSClientWinStoreJS_sample

**Comment by SandeepNayak077** [comment]
@daisukenishino 

> https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/12#issuecomment-189824205
> Please divided as follows.

Yes, sure we will divide accordingly.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

- Because following modifications in root_VS2013 has not yet been completed, please check.
  - VB version
  - Modification of [TMProtocolDefinition.xml files that is in the under of WSClient_sample folder](https://github.com/OpenTouryoProject/OpenTouryoTemplates/tree/develop/root_VS2013/programs/C%23/Samples/WS_sample/WSClient_sample/WSClientWinCone_sample).
- Merge to the past version.
  - Merge to the root.
  - Merge to root_VS2012 of Store app.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

> https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/12#issuecomment-193150009

Yes, we will soon send next set of pull requests.

> Modification of TMProtocolDefinition.xml files that is in the under of WSClient_sample folder.

We think that, this modification is not necessary in root and root_VS2012.
Because, the TMProtocolDefinition2.xml (not having project name) file is not useful for either Visual Studio development server or IIS server.

Please let us know you opinion.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

>Please let us know you opinion.

Certainly, as you say, 
you do not need to merge the TMProtocolDefinition2.xml to the old version from root_VS2012.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

> https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/12#issuecomment-193577475

Thank you for the confirmation.

> And, please tell us the current status.

We will send correspondance of VB version by today (Its in the review process).
The merge work of root and root_VS2012 is also expected to deliver by today EOD.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

The following two files will be used as follows at VS2013 or later version.

##### [TMProtocolDefinition2.xml](https://github.com/SymphonyTeleca/OpenTouryoTemplates/blob/nishino/root_VS2013/files/resource/Xml/TMProtocolDefinition2.xml)

This is new default value.

When you execute the application on IIS Express, then the virtual directory name doesn't contain the project name.

##### [TMProtocolDefinition.xml](https://github.com/SymphonyTeleca/OpenTouryoTemplates/blob/nishino/root_VS2013/files/resource/Xml/TMProtocolDefinition.xml)

When you execute the application on IIS, then the virtual directory name contains the project name.
In this case, since the default port number to use 80, you should delete the specified port number.

```XML
		<Url id="url_a" value="http://localhost:8888/ASPNETWebService/ServiceForFx.asmx"/>
		<Url id="url_b" value="http://localhost:8888/ASPNETWebService/WCFHTTPSvcForFx.svc"/>
```
▼ ▼ ▼
```XML
		<Url id="url_a" value="http://localhost/ASPNETWebService/ServiceForFx.asmx"/>
		<Url id="url_b" value="http://localhost/ASPNETWebService/WCFHTTPSvcForFx.svc"/>
```

Do not change the URI  of TCP.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

Changes to "root_VS2013" was merged to nishino branch after completing enough review.
Changes to the "root and root_VS2012" is scheduled to perform the diff check between the root folders after merge to nishino branch.

**Comment by SandeepNayak077** [comment]
@daisukenishino 

> https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/12#issuecomment-194113812

Yes we understood your concern.
We will send correspondence about this pull request.

> https://github.com/OpenTouryoProject/OpenTouryoTemplates/issues/12#issuecomment-194117352

Ok, thank you for the report.
In "root and root_VS2012", there is one changes is pending.
That is resolve URL issue in WisStore_sample.
This correspondence will be sent after completion of [this task](https://github.com/OpenTouryoProject/OpenTouryo/issues/145#issuecomment-193180002),
Because of dependency and to avoid confilcts.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

>This correspondence will be sent after completion of this task,

At first, you should carry out the corresponding URL changes for VS2013.
And, move the folder of Store App.

Otherwise, more than one issue are mixed in nishino branch.
And, they mixed in one pull request.

**Comment by daisukenishino** [comment]
@SandeepNayak077 

The following two problems have been found by comparing the develop branch and nisino branch.

- Please change "FxXMLTMProtocolDefinition" property in the appsettings to "TMProtocolDefinition2.xml" as default at VS2013 or later version.

- I think that SCDefinition2.xml is also needed like "TMProtocolDefinition2.xml" correspondence at VS2013 or later version.

Default setting should not include the project name in the URL at VS2013 or later version.
